### DISCUSSION TITLE
Upgrading from 4.0.1 to 4.1.1 - Conditional Target Frameworks

### DISCUSSION THREAD

**Discussion by mwasson74** [ardalis/GuardClauses#290] [Q&A] [answered]
After upgrading from 4.0.1 to 4.1.1, the compiler (?) is now choosing the Null extension method where the `parameterName` parameter is required instead of the definition where `parameterName` is optional.

4.0.1:
![image](https://github.com/ardalis/GuardClauses/assets/12786751/78835786-08d2-4525-b093-df6eaf1750a9)

4.1.1:
![image](https://github.com/ardalis/GuardClauses/assets/12786751/2362d9e1-4b60-48aa-962e-b32c6b7d3913)

What am I doing wrong here?

**Comment by mwasson74** [comment]
I changed my project from targeting .NET 6 to target .NET 7 and it now chooses the correct method definition. I wouldn't think that I would need to change my target version when the else branch of the [if-statement](https://github.com/ardalis/GuardClauses/blob/main/src/GuardClauses/GuardAgainstNullExtensions.cs#L26-L36) should still apply.

We have not committed to targeting .NET 7 yet because .NET 6 is the LTS version. Please advise.

**Comment by ardalis** [comment]
Yeah that's weird and unexpected. Maybe the problem (for you, at least) goes away in a couple of months when .NET 8 is LTS... :)

I'll see if I can repro the issue and work out a fix. I've updated a bunch of my projects, some net6 some net7, and haven't seen this.

**Reply by mwasson74** [reply]
@ardalis were you able to see if you can reproduce this?

**Reply by mwasson74** [reply, marked as answer]
@ardalis is my question now closed due to the fix on #303??

**Reply by ardalis** [reply]
I'm hoping so, yes. Can you confirm the latest bits are working as you expect?

**Reply by mwasson74** [reply]
Yep, now working as expected!! Thank you!!
